test/connectivity/connectivity.spec.js

```diff
@@ -0,0 +1,305 @@
+    dockerComposeFilePath,
+    expectedInstanceDetails,
+    model: testConnectionModel,
+    title: testConnectionTitle
+  }) => {
+    let wrapper;
+    let appRegistry;
+    let compassConnectStore;
+
+    before(async() => {
+      dockerComposeUp(dockerComposeFilePath);
+
+      appRegistry = new AppRegistry();
+
+      activateCompassConnect(appRegistry);
+
+      global.hadronApp = {
+        appRegistry
+      };
+
+      const ROLE = {
+        name: 'Status',
+        component: () => (<div id="statusPlugin">Status</div>)
+      };
+      global.hadronApp.appRegistry = appRegistry;
+      global.hadronApp.appRegistry.registerRole('Application.Status', ROLE);
+
+      compassConnectStore = appRegistry.getStore('Connect.Store');
+
+      // Remove all logic around saving and loading stored connections.
+      // NOTE: This is tightly coupled with the store in compass-connect.
+      // https://github.com/mongodb-js/compass-connect/blob/master/src/stores/index.js
+      compassConnectStore._saveRecent = () => {};
+      compassConnectStore._saveConnection = () => {};
+      compassConnectStore.state.fetchedConnections = [];
+      compassConnectStore.StatusActions = {
+        done: () => {},
+        showIndeterminateProgressBar: () => {}
+      };
+      compassConnectStore.appRegistry = appRegistry;
+
+      // Load the connection into our connection model.
+      const model = new Connection(testConnectionModel);
+      debug('Created model with connection string:', model.driverUrl);
+
+      // Load the connection model through compass-connect and render it.
+      // Load the connection into compass-connect's connection model.
+      compassConnectStore.state.currentConnection = model;
+      compassConnectStore.trigger(compassConnectStore.state);
+
+      // Here we use the parsed connection model and build a url.
+      // This is something that would occur if
+      // a user is switching between the connection views and editing.
+      compassConnectStore.state.customUrl = compassConnectStore.state.currentConnection.driverUrlWithSsh;
+      compassConnectStore.trigger(compassConnectStore.state);
+
+      wrapper = mount(<CompassConnectComponent />);
+
+      // Ensures the model doesn't cause any errors when rendering.
+      await compassConnectStore.validateConnectionString();
+    });
+
+    after(() => {
+      deactivateCompassConnect(appRegistry);
+
+      dockerComposeDown(dockerComposeFilePath);
+    });
+
+    it('doesn\'t have any compass connect store errors when rendering', () => {
+      expect(compassConnectStore.state.errorMessage).to.equal(null);
+      expect(compassConnectStore.state.syntaxErrorMessage).to.equal(null);
+      expect(compassConnectStore.state.isValid).to.equal(true);
+    });
+
+    context(`Connection ${testConnectionTitle} successfully connects`, () => {
+      let dataService;
+      let instanceDetails;
+      let onDataServiceConnected;
+
+      before(async() => {
```

Comment:
What is the reason to not put this in a `beforeEach` instead? Is better if to avoid interaction between executions, the cost in terms of time should not be that big.

Reply:
Oh i see the 'it' are just checking stuff on the data we fetch in the before. I guess is ok.

Reply:
I think it will make sense to move them to beforeEach if they're doing extra things on the connection, right now I think it doesn't totally make sense to reconnect for each of these tests. The cost might get higher for external connections, since it would be synchronously reconnecting/disconnecting and sometimes running a query. If it's decent latency could add up quick, but yeah not too much. I'm thinking let's move them into before each if they need guaranteed isolation. Maybe it makes sense to make them beforeEach if we want to be fully testing the disconnect ability of compass as well 🤔